templates/cc/string.go

```diff
@@ -135,12 +144,39 @@ const strTpl = `
 		}
 		*/}}
 	{{ else if $r.GetHostname }}
-		{{ unimplemented }}
-		{{/* TODO(akonradi) implement hostname constraints
-		if err := m._validateHostname({{ accessor . }}); err != nil {
-			return {{ errCause . "err" "value must be a valid hostname" }}
+	{
+		// Take a copy so we can lowercase it.
+		std::string value = {{ accessor . }};
+		std::transform(value.begin(), value.end(), value.begin(), [](unsigned char c){ return std::tolower(c); });
+
+		if (value.length() > 253) {
+			{{ err . "value must be a hostname, and hostname cannot exceed 253 characters" }}
 		}
-		*/}}
+
+		const std::regex dot_regex{"\\."};
+		const std::vector<std::string> domain_parts{
+			std::sregex_token_iterator(value.begin(), value.end(), dot_regex, -1),
+      std::sregex_token_iterator()
+		};
+		for (const auto &part : domain_parts) {
+			if (part.length() < 1 || part.length() > 63) {
```

Comment:
Instead of `part.length() < 1` use `!part.empty()`

Reply:
:+1: